app/controllers/sessions_controller.rb

```diff
@@ -69,24 +69,39 @@ def destroy
     end
   end
 
-  # GET /sessions/verify
-  def verify
-    respond_to do |format|
-      permission = true
-      unless params[:project_id].nil? || params[:verify_owner].nil?
-        owner_id = Project.find(params[:project_id]).user_id
-        verify = params[:verify_owner] == 'true'
-        session_destroyed = session[:user_id].nil?
-        is_admin = (!session_destroyed and User.find(session[:user_id]).admin)
-        permission = (verify && owner_id == session[:user_id]) || !verify || is_admin
+  # GET /sessions/permissions
+  def permissions
+    @permiss = []
+    # Session is null
+    if session[:user_id].nil?
+      respond_to do |format|
+        format.js {}
+        format.json { head :unauthorized }
       end
+      return
+    end
 
-      if session[:user_id].nil? || !permission
-        format.json { render json: '{}', status: :unauthorized }
-      else
-        format.json { render json: '{}', status: :ok }
+    is_admin = User.find(session[:user_id]).admin
+    @permiss.push 'save'
+
+    # Not admin, and you don't own or didn't specify a project
+    unless is_admin
```

Comment:
The reason I chose to do it this way is because I building a return value.

1. If you have no permissions, you return an empty list.
2. As you progress through the code, cases add to you permission list.

This is why I chose to order them this way. Being an admin gives you full permissions, thus it should be last.